File: checkio_js/home/javascript/backwardStringByWord.js
import assert from "assert";

function backwardStringByWord(text) {
    return text.split(' ').map(a => a.split('').reverse().join('')).join(' ')
}

console.log('Example:');
console.log(backwardStringByWord(''));

// These "asserts" are used for self-checking
assert.equal(backwardStringByWord(''), '');
assert.equal(backwardStringByWord('world'), 'dlrow');
assert.equal(backwardStringByWord('hello world'), 'olleh dlrow');
assert.equal(backwardStringByWord('hello   world'), 'olleh   dlrow');
assert.equal(backwardStringByWord('welcome to a game'), 'emoclew ot a emag');

console.log("Coding complete? Click 'Check' to earn cool rewards!");

File: checkio_js/home/python/backwardStringByWord.py


def backwardStringByWord(text):
    return  ' '.join(map(lambda x: x[::-1], text.split(' ')))

if __name__ == '__main__':
    print('Example:')
    print(backwardStringByWord(''))

#  These "asserts" are used for self-checking
assert backwardStringByWord('') == ''
assert backwardStringByWord('world') == 'dlrow'
assert backwardStringByWord('hello world') == 'olleh dlrow'
assert backwardStringByWord('hello   world') == 'olleh   dlrow'
assert backwardStringByWord('welcome to a game') == 'emoclew ot a emag'

print("Coding complete? Click 'Check' to earn cool rewards!")
